From 883cc555df870e45599fd5ed951843c332fcdafc Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org>
Date: Tue, 3 Dec 2019 09:55:02 +0200
Subject: Wildcards in exclude-vcs-ignore mode should not match slash

* src/exclist.c (info_attach_exclist): Use the FNM_FILE_NAME flag.
---
 src/exclist.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/exclist.c b/src/exclist.c
index 288f3be..9406f32 100644
--- a/src/exclist.c
+++ b/src/exclist.c
@@ -104,7 +104,8 @@ info_attach_exclist (struct tar_stat_info *dir)
 	    vcsfile->data = vcsfile->initfn (vcsfile->data);
 
 	  if (add_exclude_fp (vcsfile->addfn, ex, fp,
-			      EXCLUDE_WILDCARDS|EXCLUDE_ANCHORED, '\n',
+			      FNM_FILE_NAME|EXCLUDE_WILDCARDS|EXCLUDE_ANCHORED,
+			      '\n',
 			      vcsfile->data))
 	    {
 	      int e = errno;
-- 
cgit v1.2.1

